Tools/Scripts/hooks/pre-push

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/env {{ python }}
+VERSION = '1.0'
+
+import io
+import os
+import re
+import subprocess
+import sys
+
+if sys.stdout.isatty() and os.path.exists('/dev/tty'):
+    TTY = io.TextIOWrapper(io.FileIO(os.open('/dev/tty', os.O_NOCTTY | os.O_RDWR), 'r+'))
```

Comment:
We can't just listen to `stdin` when we prompt the user, because `git push` passes information to `pre-push` hooks via `stdin`. We need to grab the current "tty", if possible. I'm not convinced that actually _is_ possible on all machines (namely Windows), so if it isn't, we just don't prompt the user. Note that prompting the user is only necessary during publication of security content, which is a rare enough case that I don't think anyone has ever attempted to do so from a Windows machine.

Reply:
Makes sense, thank you for clarifying!